docs/userguides/compile.md

```diff
@@ -81,3 +81,31 @@ Alternatively, configure it to always happen:
 compile:
   use_dependencies: true
 ```
+
+## Settings
+
+Generally, configure compiler plugins using your `ape-config.yaml` file.
+For example, when using the `vyper` plugin, you can configure settings under the `vyper` key:
+
+```yaml
+vyper:
+  version: 0.3.10
+```
+
+You can also configure adhoc settings in Python code:
+
+```python
+from ape import compilers
+
+settings = {"vyper": {"version": "0.3.7"}, "solidity": {"version": "0.8.0"}}
+compilers.compile(
+   ["path/to/contract.vy", "path/to/contract.sol"], settings=settings
+)
+
+# Or, more explicitly:
+vyper = compilers.get_compiler("vyper", settings=settings["vyper"])
+vyper.compile("path/to/contract.vy")
```

Comment:
```
In [8]: vyper.compile("/home/doge/ape/TokenProject/contracts/Token.vy")
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[8], line 1
----> 1 vyper.compile("/home/doge/ape/TokenProject/contracts/Token.vy")

File ~/ape/ape-vyper/ape_vyper/compiler.py:329, in VyperCompiler.compile(self, contract_filepaths, base_path)
    327 contract_types = []
    328 base_path = base_path or self.config_manager.contracts_folder
--> 329 sources = [p for p in contract_filepaths if p.parent.name != "interfaces"]
    330 version_map = self.get_version_map(sources)
    331 compiler_data = self._get_compiler_arguments(version_map, base_path)

File ~/ape/ape-vyper/ape_vyper/compiler.py:329, in <listcomp>(.0)
    327 contract_types = []
    328 base_path = base_path or self.config_manager.contracts_folder
--> 329 sources = [p for p in contract_filepaths if p.parent.name != "interfaces"]
    330 version_map = self.get_version_map(sources)
    331 compiler_data = self._get_compiler_arguments(version_map, base_path)

AttributeError: 'str' object has no attribute 'parent'
```

it's looking like there's no parent in the string object being passed here.

Reply:
also failing functional test: 
```
=========================== short test summary info ============================
FAILED tests/functional/test_compilers.py::test_compile_str_path - KeyError: 'ApeContract1'
======= 1 failed, 1056 passed, 5 xfailed, 5 xpassed in 411.96s (0:06:51) =======
Error: Process completed with exit code 1.
```

Reply:
I fixed the example to wrap those in Path before calling Vyper.
And I fixed the test.
Thanks!

Reply:
I'm still getting the same error after those fixes.

Reply:
i decided CompilerAPI level should still only deal in Paths, so i fixed the example to wrap in Paths if doing it that way

Reply:
No problem this works now!